misc/toml-parser/src/main/java/io/ballerina/toml/internal/parser/TomlParser.java

```diff
@@ -389,27 +399,42 @@ private STNode parseValue() {
     private STNode parseNumericalNode() {
         STNode sign = parseSign();
         STNode token = parseNumericalToken();
-        SyntaxKind kind;
-        if (token.kind == SyntaxKind.DECIMAL_INT_TOKEN) {
-            kind = DEC_INT;
+        SyntaxKind kind = getNodeKindNumericNode(token.kind);
+        return STNodeFactory.createNumericLiteralNode(kind, sign, token);
+    }
+
+    private SyntaxKind getNodeKindNumericNode(SyntaxKind tokenKind) {
+        if (tokenKind == SyntaxKind.DECIMAL_INT_TOKEN) {
+            return DEC_INT;
+        } else if (tokenKind == HEX_INTEGER_LITERAL_TOKEN) {
+            return HEX_INT;
+        } else if (tokenKind == OCTAL_INTEGER_LITERAL_TOKEN) {
+           return OCT_INT;
+        } else if (tokenKind == BINARY_INTEGER_LITERAL_TOKEN) {
+            return BINARY_INT;
         } else {
-            kind = FLOAT;
+            return FLOAT;
         }
-        return STNodeFactory.createNumericLiteralNode(kind, sign, token);
     }
 
     private STNode parseNumericalToken() {
         STToken token = peek();
-        if (token.kind == SyntaxKind.DECIMAL_INT_TOKEN) {
-            return consume();
-        } else if (token.kind == SyntaxKind.DECIMAL_FLOAT_TOKEN) {
+        if (isNumerticalLiteral(token)) {
             return consume();
         } else {
             recover(token, ParserRuleContext.NUMERICAL_LITERAL);
```

Comment:
Need to update the error handler for the above change. (when recovering with NUMERICAL_LITERAL ctx)